### PR TITLE
Resolve Location

### PR DESCRIPTION
**Pull request by Tentanus**
- Added resolveLocation method to ServerSettings class

### CODE REVIEW

File: include/ServerSettings.hpp [hunk 1]
```diff
@@ -33,6 +34,7 @@ class ServerSettings
 
 	ServerSettingOption identifyServerSetting(const std::string &token);
 	// TODO: void addLocationSetting(LocationSettings settings);
+	bool resolveLocation(std::string path, HTTPMethod method);
```
Comment (saladuit) [inline include/ServerSettings.hpp:37]: const std:;string &

File: src/ServerSettings.cpp [hunk 2]
```diff
@@ -70,7 +71,35 @@ ServerSettings::identifyServerSetting(const std::string &token)
 	// TODO: this might need a costum exception
 }
 
-// TODO: void addLocationSetting(LocationSettings settings);
+const std::string
+methodToString(HTTPMethod method) // TODO: change data_types in function
```
Comment (saladuit) [inline src/ServerSettings.cpp:75]: move to HTTPMethod.cpp file

File: src/ServerSettings.cpp [hunk 3]
```diff
@@ -70,7 +71,35 @@ ServerSettings::identifyServerSetting(const std::string &token)
...
+}
+
+bool ServerSettings::resolveLocation(std::string path, HTTPMethod input_method)
+{
+	for (auto &location_instance : _location_settings)
+	{
+		if (location_instance.getPath() != path)
+			continue;
+		for (auto &methods :
+			 location_instance.getValues(LocationSettingOption::AllowMethods))
```
Comment (saladuit) [inline src/ServerSettings.cpp:97]: allowed*

**Review by saladuit — CHANGES_REQUESTED** [review]
nice! Perhaps we can already implement it after you made the final changes

File: src/ServerSettings.cpp (hunk 2 above)
Comment (Tentanus) [reply src/ServerSettings.cpp:75]: Nowhere to move to. HTTPMethod is not a class but a method

**Review by Tentanus — COMMENTED** [review]
finshed, all comments resolved

**Review by saladuit — APPROVED** [review]
Well done!

**Pull request merged**